src/NumericalAlgorithms/Interpolation/SendPointsToInterpolator.hpp

```diff
@@ -28,6 +28,89 @@ struct ReceivePoints;
+///
+/// get_block_logical_coords is called by an Action of InterpolationTarget.
+///
+/// Currently two Actions call get_block_logical_coords:
+///   SendPointsToInterpolator (called by AddTemporalIdsToInterpolationTarget
+///                             and by FindApparentHorizon)
+///   InterpolationTargetVarsFromElement (called by DgElementArray)
+template <typename InterpolationTargetTag, typename DbTags,
+          typename Metavariables, typename TemporalId>
+auto get_block_logical_coords(const db::DataBox<DbTags>& box,
+                              Parallel::ConstGlobalCache<Metavariables>& cache,
+                              const TemporalId& temporal_id) noexcept {
+  const auto& domain =
+      db::get<domain::Tags::Domain<Metavariables::volume_dim>>(box);
+  return block_logical_coordinates(
+      domain, InterpolationTargetTag::compute_target_points::points(
+                  box, cache, temporal_id));
+}
+
+/// Adds the supplied block logical coordinates to the
+/// InterpolationTarget's internal DataBox, and initializes
+/// InterpolationTarget's variables storage and lists of indices
+/// corresponding to the supplied temporal_id.
+///
+/// set_up_interpolation is called by an Action of InterpolationTarget.
+///
+/// Currently two Actions call set_up_interpolation:
+///   SendPointsToInterpolator (called by AddTemporalIdsToInterpolationTarget
+///                             and by FindApparentHorizon)
+///   InterpolationTargetVarsFromElement (called by DgElementArray)
+template <typename InterpolationTargetTag, typename DbTags, size_t VolumeDim,
+          typename TemporalId>
+void set_up_interpolation(
+    const gsl::not_null<db::DataBox<DbTags>*> box,
+    const TemporalId& temporal_id,
+    const std::vector<boost::optional<
+        IdPair<domain::BlockId,
+               tnsr::I<double, VolumeDim, typename ::Frame::Logical>>>>&
+        block_logical_coords) noexcept {
+  db::mutate<Tags::IndicesOfFilledInterpPoints<TemporalId>,
+             Tags::IndicesOfInvalidInterpPoints<TemporalId>,
+             Tags::InterpolatedVars<InterpolationTargetTag, TemporalId>>(
+      box, [&block_logical_coords, &temporal_id ](
+               const gsl::not_null<
+                   std::unordered_map<TemporalId, std::unordered_set<size_t>>*>
+                   indices_of_filled,
+               const gsl::not_null<
+                   std::unordered_map<TemporalId, std::unordered_set<size_t>>*>
+                   indices_of_invalid_points,
+               const gsl::not_null<std::unordered_map<
+                   TemporalId, Variables<typename InterpolationTargetTag::
+                                             vars_to_interpolate_to_target>>*>
+                   vars_dest_all_times) noexcept {
+        // At this point we don't know if vars_dest exists in the map;
+        // if it doesn't then we want to default construct it.
+        auto& vars_dest = (*vars_dest_all_times)[temporal_id];
```

Comment:
Can this line (and comments) be moved down ~ 14 lines, to where it will be used?

Reply:
:+1: